crates/rpc/src/v02/method/estimate_fee.rs

```diff
@@ -70,16 +70,24 @@ pub async fn estimate_fee(
     let (when, pending_timestamp, pending_update) =
         base_block_and_pending_for_call(input.block_id, &context.pending_data).await?;
 
-    let result = handle
+    let mut result = handle
         .estimate_fee(
-            input.request,
+            vec![input.request],
             when,
             gas_price,
             pending_update,
             pending_timestamp,
         )
         .await?;
 
+    if result.len() != 1 {
+        return Err(
+            anyhow::anyhow!("Internal error: expected exactly one fee estimation result").into(),
+        );
+    }
+
+    let result = result.swap_remove(0);
```

Comment:
I found the `swap_remove` weird -- thought maybe it was done for a specific reason.
```suggestion
    let result = result.pop().unwrap()
```

Reply:
I've done before adding the length check -- we can just use `pop()` after checking that the length is _exactly_ one.